src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -540,6 +540,30 @@ void loop_inst::preprocess_backedge_memory() {
             GPU_DEBUG_LOG << idx << ") add back_edge mapping with SINGLE_SHARED type, backedge_mem("
                             << backedge_mem << "), initial_mem(" << initial_mem << ")" << std::endl;
         }
+
+        if (backedge_to_prim->_node != nullptr) {
+            update_backedge_exec_deps(backedge_to_prim->get_node(), backedge_from_prim->id());
+        }
+    }
+}
+
+void loop_inst::update_backedge_exec_deps(const cldnn::program_node& node, const cldnn::primitive_id& backedge_from_prim_id) {
+    // Add _exec_deps for backedge primitives
```

Comment:
Also could you please add a comment why this change is added?

Reply:
Updated! Thanks!